src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const last = this[this.length - 1];
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += separator;
+      } else {
+        result += `${this[i]}${separator}`;
+      }
```

Comment:
Also add separator only if it's not last element
```suggestion
      if (this[i] !== undefined && this[i] !== null) {
        result += this[i];
      }
```